test/cufinufft3d1_test.cu

```diff
@@ -100,10 +100,29 @@ int main(int argc, char* argv[])
 	int type = 1;
 	ier=cufinufft_default_opts(type, dim, dplan.opts);
 	dplan.opts.gpu_method=method;
-	dplan.opts.gpu_binsizex = 16;
-	dplan.opts.gpu_binsizey = 16;
-	dplan.opts.gpu_binsizez = 2;
-	dplan.opts.gpu_maxsubprobsize = 4096;
+	dplan.opts.gpu_kerevalmeth=1;
+	switch(method){
+		case 1:
+		case 2:
+			{
+				dplan.opts.gpu_binsizex = 8;
```

Comment:
Couldn't these things be folded into the default options? Is it even desirable for users to adjust these? I doubt anyone besides @MelodyShih is going to fiddle with these.

Reply:
The default method for 3d type 1 is the sub problem method (gpu_method=2) and so the default value for `gpu_binsize*` are set for that method. Since `cufinufft_default_opts` is called before the method is set, I am not sure how to do it properly. Please let me know if you have and suggestions, thanks!

Reply:
I think these things are so far in the weeds that we may want to hardcode them (with good values). I don't think anyone will start changing these. That way we can also simplify the opts structure.

If we do want to keep them, an option would be to have the “default” bin sizes (as set by `cufinufft_default_opts`) set to zero, (or negative one) which means that the hardcoded values are used.

Reply:
I think I would still like to keep them as options so that users can overwrite them once they get more familiar with the methods. Setting the options to -1 seem to be a good idea to resolve the problem! Thanks for the suggestion. I did it in commit 2a2e51d. Now, there is a check in the plan stage that if the options are -1, then some hard coded values are used (see function `set_binsize`).